File: DIP/DIP_Lab_406/Lab/353_lab_03/Lab_03_Python/HistogramEqualization.py
import cv2
import numpy as np
from matplotlib import pyplot as plt

# Read the input image (grayscale)
input_image = cv2.imread('nature.jpeg', cv2.IMREAD_GRAYSCALE)  # Replace 'your_image.jpg' with the actual image file

# Display the original image and its histogram
plt.figure(figsize=(10, 8))

# Show original image
plt.subplot(2, 2, 1)
plt.imshow(input_image, cmap='gray')
plt.title('Input Image')
plt.axis('off')

# Plot histogram of the original image
plt.subplot(2, 2, 2)
plt.hist(input_image.ravel(), bins=256, range=[0, 256])
plt.title('Histogram of Input Image')

# Perform histogram equalization
equalized_image = cv2.equalizeHist(input_image)

# Show equalized image
plt.subplot(2, 2, 3)
plt.imshow(equalized_image, cmap='gray')
plt.title('Histogram-Equalized Image')
plt.axis('off')

# Plot histogram of the equalized image
plt.subplot(2, 2, 4)
plt.hist(equalized_image.ravel(), bins=256, range=[0, 256])
plt.title('Histogram of Equalized Image')

# Show the plots
plt.tight_layout()
plt.show()
import cv2
import numpy as np
from matplotlib import pyplot as plt

# Read the input image (grayscale)
input_image = cv2.imread('your_image.jpg', cv2.IMREAD_GRAYSCALE)  # Replace 'your_image.jpg' with the actual image file

# Display the original image and its histogram
plt.figure(figsize=(10, 8))

# Show original image
plt.subplot(2, 2, 1)
plt.imshow(input_image, cmap='gray')
plt.title('Input Image')
plt.axis('off')

# Plot histogram of the original image
plt.subplot(2, 2, 2)
plt.hist(input_image.ravel(), bins=256, range=[0, 256])
plt.title('Histogram of Input Image')

# Perform histogram equalization
equalized_image = cv2.equalizeHist(input_image)

# Show equalized image
plt.subplot(2, 2, 3)
plt.imshow(equalized_image, cmap='gray')
plt.title('Histogram-Equalized Image')
plt.axis('off')

# Plot histogram of the equalized image
plt.subplot(2, 2, 4)
plt.hist(equalized_image.ravel(), bins=256, range=[0, 256])
plt.title('Histogram of Equalized Image')

# Show the plots
plt.tight_layout()
plt.show()

File: DIP/DIP_Lab_406/Lab/353_lab_03/Lab-03_MATLAB/HistogramEqualization.m
% MATLAB code for histogram equalization

% Read the input image (grayscale)
input_image = imread('flower.jpeg'); % Replace 'your_image.jpg' with the actual image file

% Convert to grayscale if it is an RGB image
if size(input_image, 3) == 3
    input_image = rgb2gray(input_image);
end

% Display the original image
figure;
subplot(2, 2, 1);
imshow(input_image);
title('Input Image');

% Plot the histogram of the original image
subplot(2, 2, 2);
imhist(input_image);
title('Histogram of Input Image');

% Perform histogram equalization
equalized_image = histeq(input_image);

% Display the histogram-equalized image
subplot(2, 2, 3);
imshow(equalized_image);
title('Histogram-Equalized Image');

% Plot the histogram of the equalized image
subplot(2, 2, 4);
imhist(equalized_image);
title('Histogram of Equalized Image');